analysis/src/Hint.ml

```diff
@@ -0,0 +1,118 @@
+open SharedTypes
+
+type inlayHintKind = Type | Parameter
+let inlayKindToNumber = function
+  | Type -> 1
+  | Parameter -> 2
+
+let locItemToTypeHint ~full:{file; package} locItem =
+  match locItem.locType with
+  | Constant t ->
+    Some
+      (match t with
+      | Const_int _ -> "int"
+      | Const_char _ -> "char"
+      | Const_string _ -> "string"
+      | Const_float _ -> "float"
+      | Const_int32 _ -> "int32"
+      | Const_int64 _ -> "int64"
+      | Const_nativeint _ -> "int")
+  | Typed (_, t, locKind) ->
+    let fromType typ =
+      typ |> Shared.typeToString
+      |> Str.global_replace (Str.regexp "[\r\n\t]") ""
```

Comment:
Why are we doing this...? Is it to "pluck" off what you want from `typeToString`?

Reply:
`typeToString` returns with newlines. 

Example:

```res
let ppp = (Ext.make, Ext.makeProps)
```

`Ext.make` typeToSring return

```
React.componentLike<
  {"align": option<string>},
  React.element,
>
```

![image](https://user-images.githubusercontent.com/16160544/180065326-4e94df35-2163-472a-9345-6ae7b91f825d.png)

![image](https://user-images.githubusercontent.com/16160544/180066275-0bc9735c-cd07-41e4-b9d4-7a2352149009.png)
**Forgot to remove empty spaces.**

Reply:
Trim whitespace fixed by https://github.com/rescript-lang/rescript-vscode/pull/453/commits/a4353efd190c4a2ef48273611b38816bfc6206fd

Reply:
Ahh right, and the newline character appears in the output if we don't remove them I guess?

I guess the cleanest option would be to have a mode which can print without line breaks. But that might not be feasible/might be to over engineer it. What do you think @cristianoc ?

Reply:
I tested a bit. I think the whitespace is better to keep. I wonder if we should do something about the potential length of the hint, as in hide it if it's too long. What do other extensions with the same capabilities do @aspeddro , do you know that?

Great work in general, this is a really neat feature that'll be great to have for us!

Reply:
rust analyzer has a config `rust-analyzer.inlayHints.maxLength`. Default is 25; https://github.com/rust-lang/rust-analyzer/pull/8177

Great suggestion I will add.

Reply:
Done b8d1ece20839119463eb69e349865514e080ae82

Reply:
Keep whitespace fixed https://github.com/rescript-lang/rescript-vscode/pull/453/commits/bd4cfdd19d329e6179f8365114a9f7985af0a8e3

Reply:
Nice!